source/simulator/core.cc

```diff
@@ -2379,19 +2379,18 @@ namespace aspect
 
         pcout << std::endl;
 
-        // update the time step size
+        // get new time step size
         // for now the bool (convection/conduction dominated)
         // returned by compute_time_step is unused, will be
         // added to statistics later
-        old_time_step = time_step;
-        time_step = std::min (compute_time_step().first,
-                              parameters.maximum_time_step);
-        time_step = termination_manager.check_for_last_time_step(time_step);
+        const double new_time_step = termination_manager.check_for_last_time_step(
+                                       std::min(compute_time_step().first,
+                                                parameters.maximum_time_step));
 
         if (parameters.convert_to_years == true)
-          statistics.add_value("Time step size (years)", time_step / year_in_seconds);
+          statistics.add_value("Time step size (years)", new_time_step / year_in_seconds);
         else
-          statistics.add_value("Time step size (seconds)", time_step);
+          statistics.add_value("Time step size (seconds)", new_time_step);
```

Comment:
Let's leave it like this for now, but it would be nice if in a follow-up patch we could change this to ensure that the time steps we list in the statistics file add up to the current time. I'll open an issue for this.